src/org/jitsi/impl/neomedia/rtp/translator/RTCPFeedbackMessageSender.java

```diff
@@ -102,87 +113,291 @@ private long getSenderSSRC()
 
     /**
      * Sends a Full Intra Request (FIR) Command to a media sender/source with a
-     * specific synchronization source identifier (SSRC).
-     * <p>
-     * <b>Warning</b>: Due to (current) limitations in
-     * <tt>RTPTranslatorImpl</tt> and/or <tt>StreamRTPManager</tt>, a FIR
-     * command with the specified <tt>mediaSenderSSRC</tt> is sent to all
-     * <tt>MediaStream</tt>s.
-     * </p>
+     * specific synchronization source identifier (SSRC) through a specific
+     * <tt>MediaStream</tt>.
      *
      * @param mediaSenderSSRC the SSRC of the media sender/source
      * @return <tt>true</tt> if a FIR command was sent; otherwise,
      * <tt>false</tt>
      */
     public boolean sendFIR(int mediaSenderSSRC)
     {
-        boolean sentFIR = false;
-        /*
-         * XXX Currently, the method effectively broadcasts a FIR i.e. sends it
-         * to all streams connected to rtpTranslator because MediaStream's
-         * method getRemoteSourceIds returns an empty list (possibly due to
-         * RED).
-         */
-        for (StreamRTPManager streamRTPManager
-                : rtpTranslator.getStreamRTPManagers())
+        boolean registerRecurringProcessible = false;
+        synchronized (kfRequesters)
```

Comment:
@gpolitis I'd love to, but so far trying to get agreement on the CLA from my employer's legal department has proved challenging.  I think my contribution above is small enough to avoid any sort of copyright dispute. :)